src/main/java/org/jabref/logic/importer/fileformat/CffImporter.java

```diff
@@ -0,0 +1,213 @@
+        String authorStr = IntStream.range(0, citation.authors.size())
+                        .mapToObj(citation.authors::get)
+                        .map((author) -> author.vals)
+                        .map((vals) -> vals.get("name") != null ?
+                                new Author(vals.get("name"), "", "", "", "") :
+                                new Author(vals.get("given-names"), null, vals.get("name-particle"),
+                                        vals.get("family-names"), vals.get("name-suffix")))
+                        .collect(AuthorList.collect())
+                        .getAsFirstLastNamesWithAnd();
+        entryMap.put(StandardField.AUTHOR, authorStr);
+
+        // Select DOI to keep
+        if (entryMap.get(StandardField.DOI) == null && citation.ids != null) {
+            List<CffIdentifier> doiIds = IntStream.range(0, citation.ids.size())
+                            .mapToObj(citation.ids::get)
+                            .filter(id -> id.type.equals("doi"))
+                            .collect(Collectors.toList());
+            if (doiIds.size() == 1) {
+                entryMap.put(StandardField.DOI, doiIds.get(0).value);
+            }
+        }
+
+        // Select SWHID to keep
+        if (citation.ids != null) {
+            List<String> swhIds = IntStream.range(0, citation.ids.size())
+                                           .mapToObj(citation.ids::get)
+                                           .filter(id -> id.type.equals("swh"))
+                                           .map(id -> id.value)
+                                           .collect(Collectors.toList());
+
+            if (swhIds.size() == 1) {
+                entryMap.put(StandardField.SWHID, swhIds.get(0));
+            } else if (swhIds.size() > 1) {
+                List<String> relSwhIds = swhIds.stream()
+                                               .filter(id -> id.split(":").length > 3) // quick filter for invalid swhids
+                                               .filter(id -> id.split(":")[2].equals("rel"))
+                                               .collect(Collectors.toList());
+                if (relSwhIds.size() == 1) {
+                    entryMap.put(StandardField.SWHID, relSwhIds.get(0));
+                }
+            }
+        }
+
+        BibEntry entry = new BibEntry(entryType);
+        entry.setField(entryMap);
+
+        List<BibEntry> entriesList = new ArrayList<BibEntry>();
+        entriesList.add(entry);
+
+        return new ParserResult(entriesList);
+    }
+
+    @Override
+    public boolean isRecognizedFormat(BufferedReader reader) throws IOException {
+
+        ObjectMapper mapper = new ObjectMapper(new YAMLFactory());
+        CffFormat citation;
+
+        try {
+            citation = mapper.readValue(reader, CffFormat.class);
+        } catch (IOException e) {
+            return false;
+        }
+
+        if (citation != null && citation.vals.get("title") != null) {
```

Comment:
I would move this within the try block and simplify this as a one liner:
return citation!=null & citation.vals.get("title) != null